sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/FileTypeSensorTest.java

```diff
@@ -86,14 +86,21 @@ public void whenProjectOutPathsNotPresent_shouldNotAddProjectInfo_shouldNotLog()
     when(settingsMock.getString("sonar.projectName")).thenReturn("FOO_PROJ");
     when(settingsMock.getString("sonar.projectKey")).thenReturn("FOO_PROJ:GUID");
     when(settingsMock.getString("sonar.projectBaseDir")).thenReturn("BASE DIR");
-    // note that below it's the singular - Path
-    when(settingsMock.getString("sonar.cs.analyzer.projectOutPath")).thenReturn("CS OutPath");
-    when(settingsMock.getString("sonar.vbnet.analyzer.projectOutPath")).thenReturn("VB OutPath");
-    // should query the `getStringArray` method, not the `getString` method like below
-    when(settingsMock.getString("sonar.cs.analyzer.projectOutPaths")).thenReturn("CS OutPaths");
-    when(settingsMock.getString("sonar.vbnet.analyzer.projectOutPaths")).thenReturn("VB OutPaths");
-    addFileToFileSystem("foo.language", Type.MAIN);
-    addFileToFileSystem("bar.language", Type.TEST);
+    // note that below it's the singular in both calls - "projectOutputPath"
+    when(settingsMock.getString("sonar.LANG_KEY.analyzer.projectOutPath")).thenReturn("foo\\.sonarqube\\out\\0");
+    when(settingsMock.getStringArray("sonar.LANG_KEY.analyzer.projectOutPath")).thenReturn(arrayOf("foo\\.sonarqube\\out\\0"));
```

Comment:
What's the purpose of this singular form? To avoid typos in property name? Then we should mock to explicitly fail if somebody queries this property, shouldn't we?

If that's not possible (in a reasonable easy way), we should add more comments that this singular property should not be read and if the test reads and uses it, it would lead to a logged message and failed test - this mechanics is hard to see (if I even got it right :) )

Reply:
very good point, I've made the tests way too complicated